### ISSUE TITLE
[CRASH] Infinite recursive call on LooperBlockEntity function

### ISSUE THREAD

**Issue by StavWasPlayZ** [StavWasPlayZ/Even-More-Instruments#10]
> Hello there! Stumbled upon your mods when was making a modpack to play with my friends
Everything was fine until my friend decided to test out each and every function the mods have and placed a looper. The world had crashed with no chance to fix (without deleting the mod, that is).
Then I decided to try out myself and also got a crash. Beside that, I have a log now, which I hereby show you:
[crash-2024-07-04_00.24.23-server.txt](https://github.com/user-attachments/files/16091133/crash-2024-07-04_00.24.23-server.txt)
I hope there is a chance to fix this, because my friend really liked this mod and I don't want to exclude it from the modpack.

_Originally posted by @1Knightey1 in https://github.com/StavWasPlayZ/Even-More-Instruments/issues/7#issuecomment-2207382760_

**Comment by StavWasPlayZ** [comment]
RE:

Hey there! Thank you for your report.

I see the version you are playing is 1.20.4. Could you perhaps try to reproduce the issue in Minecraft 1.20.1? It might be possible I did not test enough for this specific version.

Further, could I ask if you have ever migrated the mod from any previous versions of it?

Nevertheless, I shall further delve into this issue when I'm more free, perhaps next week or so.
Waiting for your update, and apologies for the inconvenience!

**Comment by StavWasPlayZ** [comment]
Actually... I understood the issue. I'm just kinda dumb.

Note for self:

[Problematic line](https://github.com/StavWasPlayZ/Even-More-Instruments-Fabric/blob/1fd1ecbfa7e818a2d90dd07a9b74d4a726292e6a/src/main/java/com/cstav/evenmoreinstruments/block/blockentity/LooperBlockEntity.java#L133)

```java
@Override
public ItemStack getTheItem() {
    // big no no
    //return getItem(0);
    return recordIn; // <-
}
```

If you're curious, the looper basically tried to figure out: "What's my current item?", and I just answered it "idk bro check your current item"

so it checked itself and itself and itself...

...Will fix this once I issue Genshin Instruments v4.0. Should be around mid-July 😅

Again, very sorry for the inconvenience!

**Comment by 1Knightey1** [comment]
Hello, thank you for explaining what could be wrong!
No, we never used your mods before (neither Fabric, to be honest), but I can try to recreate the issue on an older version, if you need to.

**Comment by StavWasPlayZ** [comment]
Yeah, I would actually greatly appreciate that!  
Could you let me know if such issue persists in either 1.20.2 or 1.20.1?

Thanks!

**Comment by StavWasPlayZ** [comment]
Should be fixed in v6.1, will release later today.
